src/Sandboxing.php

```diff
@@ -212,6 +205,10 @@ private function remove_required_amp_markup_if_not_used( Document $dom, $effecti
 			return;
 		}
 
+		if ( $dom->ampElements->length > 0 ) {
+			return;
+		}
+
```

Comment:
Added test cases for this in https://github.com/ampproject/amp-wp/pull/7288/commits/06246081b120b82114ef1bd4b344ecbc5e01945a